Documentation about dependencies. (yaml and ssl headers)

**Is your feature request related to a problem? Please describe.**

Recently fluent-bit depends on these headers.
- yaml header
  - https://github.com/fluent/fluent-bit/pull/5360
- open ssl headers
  - https://github.com/fluent/fluent-bit/pull/5874

If these headers are not installed, user will fail to build fluent-bit source.
yaml case:
```
$ cmake .. 
-- Git hash: 9ec43447b61665cb67d3b579076f878b25b01549
-- Enabling FLB_REGEX since FLB_PARSER requires
-- xxHash build type: Debug
Found posix_memalign in stdlib.h -DHAVE_POSIX_MEMALIGN_IN_STDLIB -DHAVE_POSIX_MEMALIGN
-- Plugin dirlisting enabled 
-- Plugin liana enabled [== static ==]
-- Plugin mandril enabled 
-- LINKING monkey-liana-static;
-- Could NOT find OpenSSL, try to set the path to OpenSSL root folder in the system variable OPENSSL_ROOT_DIR (missing: OPENSSL_CRYPTO_LIBRARY OPENSSL_INCLUDE_DIR) 
CMake Error at CMakeLists.txt:678 (message):
  YAML development dependencies required for YAML configuration format
  handling.

  This is a build time dependency, you can either install the dependencies or
  disable the feature setting the CMake option -DFLB_CONFIG_YAML=Off .


-- Configuring incomplete, errors occurred!
See also "/home/taka/git/fluent-bit/build/CMakeFiles/CMakeOutput.log".
See also "/home/taka/git/fluent-bit/build/CMakeFiles/CMakeError.log".
```

ssl case:
```
[ 36%] Building C object src/CMakeFiles/fluent-bit-static.dir/flb_hmac.c.o
In file included from /home/taka/git/tmp/fluent-bit/src/flb_hmac.c:18:
/home/taka/git/tmp/fluent-bit/include/fluent-bit/flb_hmac.h:22:10: fatal error: openssl/err.h: No such file or directory
   22 | #include <openssl/err.h>
      |          ^~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [src/CMakeFiles/fluent-bit-static.dir/build.make:661: src/CMakeFiles/fluent-bit-static.dir/flb_hmac.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:6802: src/CMakeFiles/fluent-bit-static.dir/all] Error 2
make: *** [Makefile:152: all] Error 2
```
**Describe the solution you'd like**
<!--- A clear and concise description of what you want to happen. -->

Add a documentation about dependencies to build from source.
https://github.com/fluent/fluent-bit/blob/master/DEVELOPER_GUIDE.md
https://docs.fluentbit.io/manual/installation/sources/build-and-install

**Describe alternatives you've considered**
<!--- A clear and concise description of any alternative solutions or features you've considered. -->

**Additional context**
<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->
<!--- Add any other context or screenshots about the feature request here. -->

Fluent-bit also depends on cmake"3". There are issues since missing documentation.
https://github.com/fluent/fluent-bit/issues/2081
https://github.com/fluent/fluent-bit/issues/1700#issuecomment-548654731
https://github.com/fluent/fluent-bit/issues/1345

--- a/README.md
+++ b/README.md
@@ -75,8 +75,8 @@ If you are interested into more details, please refer to the [Build & Install](h
 - CMake >= 3.0
 - Flex
 - Bison
-- YAML headers
-- OpenSSL headers
+- YAML library/headers
+- OpenSSL library/headers
 
 #### Linux Packages
 

--- a/README.md
+++ b/README.md
@@ -70,6 +70,14 @@ bin/fluent-bit -i cpu -o stdout -f 1
 
 If you are interested into more details, please refer to the [Build & Install](https://docs.fluentbit.io/manual/installation/sources/build-and-install) section.
 
+#### Requirements
+
+- CMake >= 3.0
+- Flex
+- Bison
+- YAML headers
+- OpenSSL headers
+
 #### Linux Packages
 
 We provide packages for most common Linux distributions:

